module/zfs/dmu_tx.c

```diff
@@ -1107,16 +1127,16 @@ dmu_tx_assign(dmu_tx_t *tx, uint64_t txg_how)
 		if (err != ERESTART || !(txg_how & TXG_WAIT))
 			return (err);
 
-		dmu_tx_wait(tx);
+		dmu_tx_wait_flags(tx, txg_how);
 	}
 
 	txg_rele_to_quiesce(&tx->tx_txgh);
 
 	return (0);
 }
 
-void
-dmu_tx_wait(dmu_tx_t *tx)
+static void
+dmu_tx_wait_flags(dmu_tx_t *tx, txg_wait_flag_t how)
```

Comment:
The incoming `how` here is the old `uint64_t txg_how` (`TXG_WAIT` and friends). That's going to require those two sets of values to match up.

Can they be merged into a single type? If not, can they at least be declared next to each other in the headers and/or reference each other in nearby comments so they are kept in sync?

(this made a code read yesterday very difficult, at least...)

Reply:
Yes, I think this is reasonable. Thanks.

Reply:
I have tried to merge them. However, it seems the set is quite different and used in a different contexts.
To not complicate this commit further. I would leave it as it is.

Reply:
I think the only potential trouble spot is the call to `dmu_tx_wait_flags` from `dmu_tx_assign`. Its not a problem for this commit I think, but will be for something I'm working on top of this. So maybe you're right; leave it for now, and once its landed I'll offer a patch to make that call have a more explicit conversion.

Reply:
Here's a commit renaming the old `TXG_` flags to disambiguate them: [robn/zfs@ccb3143](https://github.com/robn/zfs/commit/ccb314391fb6b884e17fa8f40ca4fa705c2adf03)

(this doesn't do the [change in `dmu_tx_assign`](https://github.com/openzfs/zfs/pull/11082#discussion_r1161378437), that's [robn/zfs@a58f7030(https://github.com/robn/zfs/commit/a58f703992e1b0d8b9188bbd05556711e7d3ac86) if you just want something you can cherry-pick).

I can see why you might not want to take all of this in this PR, as its sort-of unrelated. If you prefer, I'm happy to make a new PR for it after this one is merged.